modules/web/src/app/core/services/rbac.ts

```diff
@@ -58,24 +101,51 @@ export class RBACService {
     projectID: string,
     roleID: string,
     kind: string,
-    name: string
+    name: string,
+    namespace: string
   ): Observable<ClusterBinding> {
     const options = {
       headers: new HttpHeaders(),
-      body: this._getDeleteBindingBody(kind, name),
+      body: this._getDeleteBindingBody(kind, name, namespace),
     };
     const url = `${this._newRestRoot}/projects/${projectID}/clusters/${clusterID}/clusterroles/${roleID}/clusterbindings`;
     return this._http.delete<ClusterBinding>(url, options);
   }
 
-  getRoleNames(clusterID: string, projectID: string): Observable<RoleName[]> {
-    const url = `${this._newRestRoot}/projects/${projectID}/clusters/${clusterID}/rolenames`;
-    return this._http.get<RoleName[]>(url);
+  getNamespaceRoleNames(clusterID: string, projectID: string): Observable<RoleName[]> {
+    const mapKey = clusterID + '-' + projectID;
+    const request$ = merge(this._refreshTimer$).pipe(
+      switchMap(_ =>
+        this._http.get<RoleName[]>(`${this._newRestRoot}/projects/${projectID}/clusters/${clusterID}/rolenames`)
+      ),
+      distinctUntilChanged((prev, curr) => _.isEqual(prev, curr)),
+      tap(roleNames => {
+        this._namespaceRoleNameMap.set(mapKey, roleNames);
+      })
+    );
+
+    return this._namespaceRoleNameMap.has(mapKey)
+      ? request$.pipe(startWith(this._namespaceRoleNameMap.get(mapKey)))
+      : request$;
   }
 
-  getBindings(clusterID: string, projectID: string): Observable<Binding[]> {
-    const url = `${this._newRestRoot}/projects/${projectID}/clusters/${clusterID}/bindings`;
-    return this._http.get<Binding[]>(url);
+  getNamespaceBindings(clusterID: string, projectID: string): Observable<Binding[]> {
+    const mapKey = clusterID + '-' + projectID;
+    const request$ = merge(this._refreshTimer$, this._refreshNamespaceBindings$).pipe(
+      switchMap(_ =>
+        this._http.get<Binding[]>(`${this._newRestRoot}/projects/${projectID}/clusters/${clusterID}/bindings`)
+      ),
+      distinctUntilChanged((prev, curr) => _.isEqual(prev, curr)),
+      tap(namespaceBindings => this._namespaceBindingsMap.set(mapKey, namespaceBindings))
+    );
+
+    return this._namespaceBindingsMap.has(mapKey)
+      ? request$.pipe(startWith(this._namespaceBindingsMap.get(mapKey)))
+      : request$;
+  }
+
+  refreshNamespaceBindings(): void {
+    this._refreshNamespaceBindings$.next();
   }
 
   createBinding(
```

Comment:
```suggestion
  createNamespaceBinding(
```
Similarly `deleteNamespaceBinding` below?